core/org.eclipse.cdt.core.tests/parser/org/eclipse/cdt/internal/index/tests/IndexDeductionGuideTest.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Igor V. Kovalenko.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Igor V. Kovalenko - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.cdt.internal.index.tests;
+
+import static org.eclipse.cdt.core.parser.tests.ast2.CommonCPPTypes.char_;
+import static org.eclipse.cdt.core.parser.tests.ast2.CommonCPPTypes.double_;
+import static org.eclipse.cdt.core.parser.tests.ast2.CommonCPPTypes.int_;
+
+import org.eclipse.cdt.core.dom.ast.IVariable;
+import org.eclipse.cdt.core.testplugin.TestScannerProvider;
+
+import junit.framework.TestSuite;
+
+/**
+ * AST tests for C++17 deduction guides via PDOM.
+ */
+public abstract class IndexDeductionGuideTest extends IndexBindingResolutionTestBase {
+	private static void cxx17SetUp() {
+		//	// STD C++ parts
+		//	TestScannerProvider.sDefinedSymbols.put("__SIZEOF_SHORT__", "2");
```

Comment:
Thanks, I was not sure if it will have a lot of sense to somehow join AST2TestBase support for version-specific sets of macros but in the end only specific feature-detection was needed for new test. Cleaned this up now.